components/user/UserEvents.js

```diff
@@ -1,46 +1,66 @@
 import { useState } from "react";
-
 import EventCard from "@components/event/EventCard";
 import Alert from "@components/Alert";
-import EventKey from "@components/event/EventKey";
 
 export default function UserEvents({ data }) {
-  const [eventType, seteventType] = useState("future");
-  const futureEvents = data.events.filter(
-    (event) => new Date(event.date.start) > new Date()
-  );
+  const [eventType, setEventType] = useState("all");
+  
+  const eventOptions = [
+    { value: 'all' , name: 'All Events'},
+    { value: 'future' , name: 'Future Events'},
+    { value: 'ongoing', name: 'Ongoing Events'},
+    { value: 'virtual', name: 'Virtual Events'},
+    { value: 'inPerson', name: 'In-Person Events'},
+    { value: 'cfpOpen', name:'Events with open CFP'},
+    { value: 'past', name: 'Past Events'}
+  ]
 
-  let categorisedEvents = {
-    future: futureEvents,
-    ongoing: data.events.filter(
-      (event) =>
-        new Date(event.date.start) < new Date() &&
-        new Date(event.date.end) > new Date()
-    ),
-    virtual: futureEvents.filter((event) => event.isVirtual === true),
-    inPerson: futureEvents.filter((event) => event.isInPerson === true),
-    cfpOpen: futureEvents.filter((event) =>
-      event.date.cfpClose ? new Date(event.date.cfpClose) > new Date() : false
-    ),
-    past: data.events
-      .filter((event) => new Date(event.date.end) < new Date())
-      .sort((a, b) => new Date(b.date.start) - new Date(a.date.start)),
+  const handleEventTypeChange = (event) => {
+    setEventType(event.target.value);
   };
 
+  const eventsToShow = eventType === "all"
+    ? data.events
+    : data.events.filter(event => {
+        const startDate = new Date(event.date.start);
+        const endDate = new Date(event.date.end);
+        const now = new Date();
+        return eventType === "future" && startDate > now
+          || eventType === "ongoing" && startDate <= now && now <= endDate
+          || eventType === "virtual" && startDate > now && event.isVirtual
+          || eventType === "inPerson" && startDate > now && event.isInPerson
+          || eventType === "cfpOpen" && startDate > now && event.date.cfpClose > now
+          || eventType === "past" && endDate < now;
+      });
+
   return (
-    <div className="mt-6">
-      <EventKey
-        categorisedEvents={categorisedEvents}
-        onToggleEventType={(newValue) => seteventType(newValue)}
-      />
+    <div className="m-6">
+      <label
+        htmlFor="event-type-select"
+        className=" mr-2 font-semibold text-gray-700"
+      >
+        Select Event Type:
+      </label>
+      <select
```

Comment:
We have `form/Input.js`, I think the `select` could be moved there also to its own component